### DISCUSSION TITLE
How to disable calling the camera in H5 iOS

### DISCUSSION THREAD

**Discussion by SmileLiyan** [community/community#137240] [Programming Help]
### Select Topic Area

Question

### Body

How to disable calling the camera in H5 iOS,please help me

Or how to set up an iOS gallery through JavaScript code

**Comment by SajanGhimire1** [comment]
1. Remove getUserMedia() Calls: Check and remove any code that uses navigator.mediaDevices.getUserMedia().
2. Avoid capture="camera": Remove the capture attribute from <input> tags.

3. Set Permissions-Policy Header:
Permissions-Policy: camera=()

4. Handle Permissions - JavaScript:
navigator.permissions.query({ name: 'camera' }).then(status => {/* Handle access */});

5. Disable in Safari Settings: Go to Settings > Safari > Camera > Set to Deny.
--This will block camera access in your H5 iOS app

**Reply by SmileLiyan** [reply]
What I mean is to use H5 and JavaScript code

**Comment by SajanGhimire1** [comment]
***Once follow this***
Disable Camera Access with JS:
``navigator.mediaDevices.getUserMedia` = function() {
  return Promise.reject(new Error("Camera access is disabled."));
};
``

Avoid Using Camera with Input Tags:
`<input type="file" accept="image/*">`

Set Up an iOS Gallery with JS:
```<input type="file" accept="image/*" onchange="previewImage(event)">
<div id="gallery"></div>
<script>
function previewImage(event) {
  const files = event.target.files;
  const gallery = document.getElementById('gallery');
  gallery.innerHTML = '';

  Array.from(files).forEach(file => {
    const img = document.createElement('img');
    img.src = URL.createObjectURL(file);
    img.style.maxWidth = '100px';
    gallery.appendChild(img);
  });
}
</script>``


now  will prevent the camera from being accessed and allow users to select images from the iOS gallery using JavaScript.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @SmileLiyan ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
